sympy/combinatorics/pc_groups.py

```diff
@@ -0,0 +1,239 @@
+                        break
+        return exponent
+
+
+def minimal_uncollected_subwords(word, relative_order):
+    """
+    Returns the minimal uncollected subwords.
+
+    Examples
+    ========
+    >>> from sympy.combinatorics.pc_groups import minimal_uncollected_subwords
+    >>> from sympy.combinatorics.free_groups import free_group
+
+    Example 8.7 Pg. 281 from Handbook
+    >>> F, x1, x2 = free_group("x1, x2")
+    >>> word = x2**2*x1**7
+    >>> relative_order = [3, 2]
+    >>> minimal_uncollected_subwords(word, relative_order)
+    {x1**7: 1, x1*x2**2: 0}
+
+    """
+    array = word.array_form
+    uncollected_subwords = {}
+    for i in range(len(array)-1):
+        if array[i+1][1] > 0:
+            # case-1:  v = x[i]**a*x[i+1]
+            uncollected_subwords[array[i][0]**array[i][1]*array[i+1][0]] = 0
+        else:
+            # case-2: v = x[i]**a*x[i+1]*-1
+            uncollected_subwords[array[i][0]**array[i][1]*array[i+1][0]**-1] = 0
+
+        if all(array[i][1]!=exp for exp in range(relative_order[i])):
+            # case-3: v = x[i]**a
+            uncollected_subwords[array[i][0]**array[i][1]] = 1
+
+    i = len(array)-1
+    if all(array[i][1]!=exp for exp in range(relative_order[i])):
+        # case-3: v = x[i]**a
+        uncollected_subwords[array[i][0]**array[i][1]] = 1
+
+    return uncollected_subwords
+
+def _relations(relators):
+    """
+    Separates the given relators of pc presentation in power and
+    conjugate relations.
+
+    Examples
+    ========
+    >>> from sympy.combinatorics.pc_groups import _relations
+    >>> from sympy.combinatorics.free_groups import free_group
+    >>> F, x1, x2 = free_group("x1, x2")
+    >>> pc_relators = {x1**2 : 1, x1*x2*x1**-1 : x2**-1, x1**-1*x2*x1 : x2**-1}
+    >>> power_rel, conj_rel = _relations(pc_relators)
+    >>> power_rel
+    {x1**2: 1}
+    >>> conj_rel
+    {x1**-1*x2*x1: x2**-1, x1*x2*x1**-1: x2**-1}
+
+    """
+    power_relators = {}
+    conjugate_relators = {}
+    for key, value in relators.items():
+        if len(key.array_form) == 1:
+            power_relators[key] = value
+        else:
+            conjugate_relators[key] = value
+    return power_relators, conjugate_relators
+
+def _reduce_word(word, power_relators):
+    """
+    Reduce the given word with the help of power
+    relators only.
+
+    Examples
+    ========
+    >>> from sympy.combinatorics.pc_groups import _reduce_word
+    >>> from sympy.combinatorics.free_groups import free_group
+    >>> F, x1, x2 = free_group("x1, x2")
+    >>> power_relators = {x1**2: 1}
+    >>> word = x1**7
+    >>> _reduce_word(word, power_relators)
+    x1
+
+    """
+    group = word.group
+    dividend = word.array_form[0][1]
+    rem = dividend
+    for w, v in power_relators.items():
+        divisor = w.array_form[0][1]
+        if w.letter_form_elm[0] == word.letter_form_elm[0] and dividend % divisor < rem:
+            rem = dividend % divisor
```

Comment:
`//` and `%` should not be used with noncommutative group elements.

Reply:
Can't we use these for exponents of elements?

Reply:
Oh yes. I didn't notice that they were integers.